src/test/java/com/example/cats/controller/ProductControllerTestIT.java

```diff
@@ -0,0 +1,87 @@
+package com.example.cats.controller;
+
+import com.example.cats.domain.Category;
+import com.example.cats.dto.ProductDTO;
+import com.example.cats.service.ProductService;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@WebMvcTest(ProductController.class)
+class ProductControllerTestIT {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ProductService service;
+
+    @TestConfiguration
+    static class MockConfig {
```

Comment:
that's not how the tests are done, so it's worng, I will do - in the integration testing part